keyvault/certrefresher_test.go

```diff
@@ -0,0 +1,127 @@
+package keyvault
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+)
+
+func TestCertRefresher(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
I have a little helper function that might be useful here:

```go
func testContext(t *testing.T) (context.Context, context.CancelFunc) {
	ctx := context.Background()
	if deadline, ok := t.Deadline(); ok {
		// a timeout was set, so use that as the context's deadline
		return context.WithDeadline(ctx, deadline)
	}
	return context.WithCancel(ctx)
}
```

So the usage here would look like:

```go
ctx, cancel := testContext(t)
defer cancel()
```

It's similar, but it'll use the deadline set in the `*testing.T` if a `-timeout` flag was provided.

Reply:
added this, it's nice convenience function for any test, so added a todo to find a better home for it in the future

Reply:
@ramiro-gamarra  Yeah, we'll have to chat about a good home for it in DNC as well, since I made a similar suggestion on a PR over there :)